presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2369,9 +2372,26 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
                 .orElseThrow(() -> new TableNotFoundException(tableName));
 
         Optional<HiveBucketHandle> hiveBucketHandle = getHiveBucketHandle(table);
+
         if (!hiveBucketHandle.isPresent()) {
```

Comment:
So this case is basically, it hasn't already been bucketed for us by the user, so we are going to pretend it's bucketed when writing out the files?

Reply:
Right. If the table is bucketed we have to follow the table bucketing. No other way round.

Now I rethink about it, maybe distinguish between  table data partitioning and table write shuffle partitioning (or some other name) might actually make code easier to understand. Otherwise -- why it's actually not partitioned by XX but we pretend them to be partitioned by XX?  . And this "if bucketed, use bucketing, otherwise, use partition column" can be a bit difficult to understand and maintain. 

cc @arhimondr